### DISCUSSION TITLE
What third party bots, tools, or integrations would you like to see?

### DISCUSSION THREAD

**Discussion by gabek** [owncast/owncast#712] [Ideas]
While still not yet ready for prime time, we're playing with Owncast 3rd party APIs for the first time.  I wanted to start a discussion on what people might want to see people build that would integrate with Owncast.

There's so many bots, integrations, and tools that people use on Twitch.  Would any of them be good candidates to live as an Owncast equivalent?  I know I'd love to see some kind of [Nightbot](https://docs.nightbot.tv/commands/commandlist) for Owncast or [Stream Elements](https://streamelements.com/features/chatbot).  While obviously these are rather large products, it'd be cool to hear about what specific features people are into.

I'll start with some ideas I've had:

1. Simple reply bot.  I see on Twitch a lot people having a pre-set list of commands to answer questions and give links.  For example `!about` gives a blurb about the stream and a link to their home page. (Using inbound chat API)
2. `!nowplaying` bot that responds with a link to the current song.  Could use LastFM or the API of whatever you use to listen to music.
3. Get notified via email or text message when a streamer goes live. (Using outbound webhook)
4. Accept donations and have a message go into the chat. (Using inbound chat API).
5. `!random` pick a random chatter to win something (Use the connected clients API to get a list of users).

**Comment by nebunez** [comment]
1. `!dice`, `!roll` - either a predetermined or user-determined (`!roll 1d6`) random dice roll
2. A chat betting system - A streamer playing Fortnite allows chat to bet on what place he will finish the game in. This would work best with persistent user accounts.
3. A 'quick' poll system - Streamer can start a poll and allow viewers to respond to poll choices. Easy implementation - use the connected clients API to track poll responses; this could be exploited easily, though. Better implementation - used with persistent user accounts.
4. Trivia - When trivia is activated, a question is posted in chat, users respond with a command prefix (`!answer`) and either first correct answer, or after N seconds, the question is over and correct answer revealed. A points system could be implemented with persistent user accounts, but the basic doesn't really need that.
5. `!gif` (e.g., Giphy) support

**Comment by controlfreakstudio** [comment]
ability to stream to owncast using the OBS recording output ffmpeg ala icecast. Thus allowing the primary stream to go to restream for corporate options and not costing the using 19+ bones per month to open up a custom single other option for restream

**Reply by jeyemwey** [reply]
Hi, I think that an icecast ingress is a larger topic that's not in the scope of these _integrations_. If you don't want to pay restream, checkout [`prism`](https://github.com/muesli/prism).

**Comment by drizzt09** [comment]
This is 3 years old. Did any get implemented?

I would like a bot that auto answers certain repeat questions. 

Also custom quick triggers to provide common information.

Also scheduled repeat messages. 

All within chat. Not on the stream.

**Reply by gabek** [reply]
There are all kinds of things people have built. I can't point to all of them, but there's a few linked at https://owncast.online/examples/projects/. You might find some more by searching GitHub for Owncast.

**Comment by drizzt09** [comment]
Some of these don't have any instructions.  Here is the code. Figure it out
yourself I guess.

**Reply by gabek** [reply]
You should contact the people who built them if you have questions. Owncast did not build any of them.
